moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame.cpp

```diff
@@ -199,6 +199,12 @@ MotionPlanningFrame::~MotionPlanningFrame()
 
 void MotionPlanningFrame::approximateIKChanged(int state)
 {
+  // This is needed to prevent UI event (resuming the options) triggered
+  // before query_start_state_ and query_goal_state_ is ready
+  if (first_time_)
+  {
+    return;
+  }
   planning_display_->useApproximateIK(state == Qt::Checked);
```

Comment:
What happens, if this is called when `first_time_` is `true`?

Reply:
https://github.com/ros-planning/moveit/blob/87b9a3592e16d11e76dc5b041ab611dcdf5a5381/moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_display.cpp#L949-L963

`query_start_state_` and `query_goal_state_` would be null (tested), no side effects.

Signals triggered before these "states" are initialized (in this case, `MotionPlanningDisplay::load()`) would not be actually executed, making the UI state not consistent with the KinematicsQueryOptions.

Because I manually triggered a "sync" in `MotionPlanningFrame::changePlanningGroupHelper()`, this is just for not executing `planning_display_->useApproximateIK()` twice. Should I remove this check?

Reply:
Yes, I would drop this additional check.

Reply:
@rhaschke Got it. Has force-pushed the changes.